src/components/EditableField.tsx

```diff
@@ -8,31 +8,61 @@ type Props = {
   initialValue: string;
   type: HTMLInputTypeAttribute;
   onSave: (value: string) => void;
+  placeholder: string;
 };
 
-const EditableField = ({ label, initialValue, type, onSave }: Props) => {
+const EditableField = ({
+  label,
+  initialValue,
+  type,
+  onSave,
+  placeholder,
+}: Props) => {
   const [value, setValue] = useState(initialValue);
+  const [wizard, setWizard] = useState<
+    | {
+        type: "initial";
+      }
+    | {
+        type: "loading";
+      }
+    | {
+        type: "editable";
+      }
+    | {
+        type: "error";
+        message: string;
+      }
+  >({ type: "initial" });
 
-  const [isEditable, setIsEditable] = useState(false);
-
-  const onSubmit = (event: { preventDefault: () => void }) => {
+  const onSubmit: React.FormEventHandler<HTMLFormElement> = async (event) => {
     event.preventDefault();
-    setIsEditable(false);
-    onSave(value);
+    try {
+      setWizard({ type: "loading" });
+      await onSave(value);
+      setWizard({ type: "initial" });
+    } catch (err) {
+      setWizard({ type: "error", message: "Unable to change your name." });
```

Comment:
Why are you "swallowing" the error? You should:

1. Use the `message` from the `err` object, if it has one
2. Otherwise, default to that custom message you added